extensions/amp-intaker-chat-widget/0.1/widget.js

```diff
@@ -0,0 +1,570 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const Intaker = {};
+/**
+ *
+ * @constructor
+ */
+Intaker.Widget = function() {
+  let CookiesAPI = window['CookiesApi'];
+  let chatIsActive = false;
+  let previewChatIsActive = false;
+  const frameId = 'chatter-bot-iframe';
+  const frameContainerId = 'chatter-bot-frame-container';
+  const btnId = 'chatter-bot-launcher';
+  const closeBtnId = 'chatter-bot-widget-close';
+  const restartBtnId = 'chatter-bot-launcher-restart';
+  const INTAKER_CHAT_URL = btoa('INTAKER_CHAT_URL');
+  const INTKER_CHAT_URL = btoa('INTKER_CHAT_URL'); //support for old generated widgets
+  let chatUrlHash = ''; //window[INTAKER_CHAT_URL] || window[INTKER_CHAT_URL];
+  let url = ''; //atob(chatUrlHash);
+  let useQA = window['USE_INTAKER_QA'];
+  let frameContainer, closeBtn, restartBtn, lunchBtn, frame;
+  let api = '';
+  const cssUrl =
+    (useQA
+      ? 'https://intakerclientqa.azurewebsites.net'
+      : 'https://intaker.co') + '/dist/chat-widget.min.css';
+  let directLink = '';
+  const isDesktop = !isMobile();
+  let cookieName = 'INTAKER_CHAT_WIDGET_';
+  const cookieValue = {};
+  const DEFAULT_AVATAR =
+    'https://intaker.blob.core.windows.net/dashboard/a6.png';
+  let dingSound = null;
+  let launcherContainer = null;
+  let isUniqueVisit = false;
+  let externalUrl = '';
+  const originalTitle = document.title || '';
+  let isAMP = false;
+  let platform = null;
+  let Templates = window['IntakerWidgetTemplates'];
+  let chatUniqueId = null;
+  let manualOpenedChat = false;
+
+  /**
+   *
+   * @param {Element} el
+   * @param {string} name
+   * @return {Element}
+   */
+  function addClass(el, name) {
+    el.classList.add(name);
+    return el;
+  }
+
+  /**
+   *
+   * @param {Element} el
+   * @param {string} name
+   * @return {Element}
+   */
+  function removeClass(el, name) {
+    el.classList.remove(name);
+    return el;
+  }
+
+  /**
+   *
+   * @param {string} url
+   */
+  function loadCss(url) {
+    const link = document.createElement('link');
+    // link.id    = 'chatter-bot-css';
+    link.rel = 'stylesheet';
+    link.type = 'text/css';
+    link.href = url;
+    link.media = 'all';
+    (document.getElementsByTagName('head')[0] || document.body).appendChild(
+      link
+    );
+  }
+
+  /**
+   *
+   * @param {string} string
+   */
+  function addToBody(string) {
+    let tmp = document.createElement('div');
+    tmp./*OK*/ innerHTML = string;
```

Comment:
This isn't OK. It needs to be clearly visible here that this isn't an XSS vector.

Reply:
I make the html templates in html files and the use `grunt-contrib-htmlmin` and `grunt-html-convert` to convert them into JS variables https://github.com/ampproject/amphtml/pull/24337/files#diff-5030b4bdd01fef76b8e8d43db552fdbeR16

How should I change this?

Reply:
usnig sanitizeHtml now